src/client/index.ts

```diff
@@ -40,49 +40,40 @@ export default function Client<
     queue[id] = resolve;
   };
 
-  const deprecatedProxyHandler = {
-    get(_: any, procedureName: any) {
-      return function (...params: unknown[]) {
-        const id = `${new Date().valueOf()}-${procedureName.toString()}-${JSON.stringify(
-          params
-        )}`;
-
-        socket.emit("function-call", {
-          procedureName,
-          params,
-          id,
-        });
-
-        return new Promise((resolve) => waitForResult(id, resolve));
-      };
-    },
-  };
-
-  function LogProxy(path: string): unknown {
-    return new Proxy(() => {}, {
-      get: function (_, prop) {
-        return LogProxy(`${path ? `${path}.` : ""}${String(prop)}`);
-      },
-      apply: function (_, __, argumentsList) {
-        console.info(
-          `RocketRPC Clint Info: Called function at path: ${path} with parameters: ${argumentsList}`
-        );
-        const id = `${new Date().valueOf()}-${path}-${JSON.stringify(
-          argumentsList
-        )}`;
-
-        const functionCallParams: FunctionCallParams = {
-          id,
-          procedurePath: path,
-          params: argumentsList,
-        };
-
-        socket.emit("function-call", functionCallParams);
-
-        return new Promise((resolve) => waitForResult(id, resolve));
-      },
-    });
+  function LogProxy(path: string, options: { socket?: Socket } = {}): unknown {
+    return new Proxy(
+      {},
+      {
+        get: function (_, prop) {
+          if (prop === "socket") {
```

Comment:
This sort of name-spaces away `socket` on the client-side.

This will prevent our users from accessing the `socket` keyword in the client right? Since it will always resolve to the `socket-io` client.

Reply:
This could probably be

```suggestion
          if (prop === "socket" && path === "") {
```

Reply:
Also, should this prop be named `ROCKETRPC_SOCKET_CLIENT` or something?